src/shared/openssl-util.c

```diff
@@ -137,4 +223,159 @@ int string_hashsum(
 
 }
 #  endif
+
+/* We use derived shared secret as volume key for EC keys.
+ * The following is the procedure.
+ * First we generate a pair of EC key of the same curve group.
+ * Then we use newly generated private key to derive shared secret
+ * with the public key in the token.
+ * Finally, we simplely forget the private key, store the public key
+ * to pkcs11-key field of LUKS token and use the dervied shared secret
+ * as the volume key (we don't apply any KDF here since LUKS will apply
+ * PBKDF on the key).
```

Comment:
can you line break this to 109ch width. That is what we usually use, and what all the vim/emacs/editorconfig/llvm-format metadata files in our tree also indicate. Please stick to that.